src/error.c

```diff
@@ -152,8 +152,16 @@ Obj FuncCURRENT_STATEMENT_LOCATION(Obj self, Obj context)
     return retlist;
 }
 
-Obj FuncPRINT_CURRENT_STATEMENT(Obj self, Obj context)
+Obj FuncPRINT_CURRENT_STATEMENT(Obj self, Obj stream, Obj context)
 {
+    /* HACK: we want to redirect output */
+    int openedOutput = 1;
+    if ((IsStringConv(stream) && !OpenOutput(CSTR_STRING(stream))) ||
+        (!IS_STRING(stream) && !OpenOutputStream(stream))) {
+        Pr("Can not open output stream\n", 0L, 0L);
+        openedOutput = 0;
+    }
+
```

Comment:
So when you did not manage to open the output, you still print an error message to the current output and set a flag? 

Did you try just raising an error using `ErrorMayQuit`?

Reply:
I thought calling other Error functions may lead to an infinite loop. `ErrorMayQuit` itself also calls `ErrorInner`.

I will change it to print to `*errout*` - or quit if that also fails - to prevent that GAP could silently write to e.g. a file when encountering errors.

Reply:
Is there a better option than quitting if I also can't open *errout*?

Reply:
Ah, I can tell ErrorMayQuit to not go into a break loop. I'll try that.

Reply:
If you can't open errout, I would just exit. For example, the C level function PRINT_OR_APPEND_TO, which is used for printing to streams, if it tries to print to errout and fails, just exits GAP.

Reply:
gah, the two arguments to `ErrorMayQuit` are forwarded to `Pr`. I thought they were arguments that toggle the behaviour. :/

Reply:
Ah, thanks for the hint.